File: ArrayMaxAny/ArrayMaxAny.js
const ages = [10, 6, 15, 93, 42, 7, 32];

const findMaxUnderBoundary = (inputArray, topBoundary) => {
    let currentMax = Number.MIN_VALUE;

    for (let i = 0; i < inputArray.length; i++) {
        if (inputArray[i] < topBoundary) {
            currentMax = Math.max(currentMax, inputArray[i])
        }
    };

    return currentMax;
};

const findTopElement = (inputArray, numberOfElements) => {
    const topElements = [];

    let previousMax = Number.MAX_VALUE;

    for (let i = 0; i < numberOfElements; i++) {
        let currentMax = findMaxUnderBoundary(inputArray, previousMax);
        previousMax = currentMax;
        topElements[i] = currentMax;
    }

    return topElements;
}

let top3age = findTopElement(ages, 3)

console.log(top3age);

File: ArrayMaxAny/ArrayMaxAny.java
package ArrayMaxAny;
// Поиск заданного количества максимальных значений в массиве

import java.util.Arrays;

public class ArrayMaxAny {

    public static int findMaxUnderBoundary(int[] inputArray, int topBoundary) {
        int currentMax = Integer.MIN_VALUE;
        for(int i = 0; i < inputArray.length; i++) {
            if (inputArray[i] < topBoundary) {
                currentMax = Math.max(currentMax, inputArray[i]);
            }
        }
        return currentMax;
    }

    public static int[] findTopElement(int[] inputArray, int numberOfElements) {
        int [] topElements = new int[numberOfElements];

        int previousMax = Integer.MAX_VALUE;

        for(int i = 0; i < numberOfElements; i++) {
            int currentMax = findMaxUnderBoundary(inputArray, previousMax);
            previousMax = currentMax;
            topElements[i] = currentMax;
        }

        return topElements;
    }
    public static void main(String[] args) {
        int[] ages = {10, 6, 15, 93, 42, 7, 32};
        
        int[] top3ages = findTopElement(ages, 3);
        System.out.println(Arrays.toString(top3ages));
    }
}